test/unit/specs/mount/options/mocks.spec.js

```diff
@@ -84,4 +84,46 @@ describe('mount.mocks', () => {
     const freshWrapper = mount(Component)
     expect(typeof freshWrapper.vm.$store).to.equal('undefined')
   })
+
+  it('replaces lifecycle hooks with mocks', () => {
+    const LIFECYCLE_HOOKS = [
+      'beforeCreate',
+      'created',
+      'beforeMount',
+      'mounted',
+      'beforeUpdate',
+      'updated',
+      'beforeDestroy',
+      'destroyed',
+      'activated',
+      'deactivated'
+    ]
+
+    const originalHooks = {}
+    const mockedHooks = {}
+    LIFECYCLE_HOOKS.forEach(hook => {
+      originalHooks[hook] = sinon.spy()
+      mockedHooks[hook] = sinon.spy()
+    })
+
+    const wrapper = mount({
+      ...originalHooks
+    }, {
+      mocks: mockedHooks
+    })
+
+    wrapper.setData({})
+
+    // call methods that will not be triggered by mount, setData and destroy manually
+    wrapper.vm.updated() // only called in v2.0.8
```

Comment:
Why is it called in 2.0.8?

Reply:
It would be good to figure out why it does this, or add a warning if a user is using Vue 2.0.x if they try to stub updated.

Reply:
Yeah, it is poorly phrased.
I wanted to simulate behavior that would call each lifecycle-hook exactly once, but my Vue-knowledge is just not deep enough to do so. So I ended up using this combination of `mount`, `setData` and `destroy` to get as many as possible called.
What happened now is that the lifecycle-hooks `activated`, `deactivated` and `updated` were not called, that is why I had to call them manually to verify if the replacing spies are used.
Now in Vue 2.0.8 the `updated` method gets called, even if there is no actual change, in > 2.1 it does not get called, I guess it compares the virtual dom to figure out that nothing gets updated and therefore does not call that lifecycle hook. In 2.4.2 (my local version) it did not get called, but in the `test:compatibility` it also runs on v2.0.8 and therefore the hook gets called in the test. I still call it 'manually' via `wrapper.vm.updated()`, and need to check if it was called once or twice (line 126).

It might be a bit overkill, to keep it simple we could just call all methods directly and check on `called` and `notCalled`, if you prefer this. It was my first approach as I like to make the tests as specific as possible, therefore the `calledOnce` and the poorly-phrased comment. I will expand the comment a bit to make it understanding this issue easier.